src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -0,0 +1,1063 @@
+    }
+};
+
+template<> struct VectorConverter<2, Double, ufbx_vec2> {
+    static Vector<2, Double> from(const ufbx_vec2& other) {
+        return {other.x, other.y};
+    }
+};
+
+template<> struct VectorConverter<3, Double, ufbx_vec3> {
+    static Vector<3, Double> from(const ufbx_vec3& other) {
+        return {other.x, other.y, other.z};
+    }
+};
+
+template<> struct VectorConverter<4, Double, ufbx_vec4> {
+    static Vector<4, Double> from(const ufbx_vec4& other) {
+        return {other.x, other.y, other.z, other.w};
+    }
+};
+
+template<> struct QuaternionConverter<Double, ufbx_quat> {
+    constexpr static Quaternion<Double> from(const ufbx_quat& other) {
+        return {{other.x, other.y, other.z}, other.w};
+    }
+};
+
+template<> struct RectangularMatrixConverter<4, 3, Double, ufbx_matrix> {
+    constexpr static RectangularMatrix<4, 3, Double> from(const ufbx_matrix& other) {
+        return RectangularMatrix<4, 3, Double>{
+            Vector3d(other.cols[0]),
+            Vector3d(other.cols[1]),
+            Vector3d(other.cols[2]),
+            Vector3d(other.cols[3]),
+        };
+    }
+};
+
+}}}
+
+namespace Magnum { namespace Trade {
+    
+using namespace Containers::Literals;
+
+namespace {
+
+struct MeshChunk {
+    uint32_t meshId;
+    uint32_t meshMaterialIndex;
+};
+
+ufbx_load_opts loadOptsFromConfiguration(Utility::ConfigurationGroup& conf) {
+    static_cast<void>(conf);
+
+    ufbx_load_opts opts = { };
+
+    opts.generate_missing_normals = conf.value<bool>("generateMissingNormals");
+    opts.strict = conf.value<bool>("strict");
+    opts.disable_quirks = conf.value<bool>("disableQuirks");
+    opts.load_external_files = conf.value<bool>("loadExternalFiles");
+    opts.ignore_geometry = conf.value<bool>("ignoreGeometry");
+    opts.ignore_animation = conf.value<bool>("ignoreAnimation");
+    opts.ignore_embedded = conf.value<bool>("ignoreEmbedded");
+    opts.ignore_all_content = conf.value<bool>("ignoreAllContent");
+
+    /* We need to split meshes by material so create a dummy ufbx_mesh_material
+       containing the whole mesh to make processing code simpler. */
+    opts.allow_null_material = true;
+
+    return opts;
+}
+
+inline Int typedId(ufbx_element *element) {
+    return element ? (Int)element->typed_id : -1;
+}
+
+inline SamplerWrapping toSamplerWrapping(ufbx_wrap_mode mode) {
+    if (mode == UFBX_WRAP_CLAMP) {
+        return SamplerWrapping::ClampToEdge;
+    } else if (mode == UFBX_WRAP_REPEAT) {
+        return SamplerWrapping::Repeat;
+    } else {
+        /* @todo: What to do about unhandled enums */
+        return SamplerWrapping::Repeat; /* LCOV_EXCL_LINE */
+    }
+}
+
+inline void logError(const char *prefix, const ufbx_error &error) {
+    if (error.info_length > 0) {
+        Error{} << prefix << Containers::StringView(error.description) << " (" << error.info << ")";
+    } else {
+        Error{} << prefix << Containers::StringView(error.description);
+    }
+}
```

Comment:
I had to use explicit `Containers::StringView()` here as otherwise `operator<<` was ambiguous, should I add some stream functionality to `ufbx_string` as well?

Reply:
Ah because now it can convert to both `String` and `StringView`, right? :sweat_smile: The cast is fine I think, and I should fix the silly ambiguity on my side instead.

Reply:
Yea exactly, but the one cast is not too bad so no worries :D